docs/adapters/adapter_api.md

```diff
@@ -0,0 +1,162 @@
+# Juttle Adapter API
+
+This document describes the low level API used to implement juttle adapters.
```

Comment:
Does it make sense to link from here to https://github.com/juttle/juttle-adapter-template as an example of implementation that uses the API? And perhaps to the gmail adapter if it's still meant to be a "teaching adapter"?